### ISSUE TITLE
ROI: strip lines in literalinclude

### ISSUE THREAD

**Issue by andreacassioli** [sphinx-doc/sphinx#2029]
Hi,
I have greatly appreciated the improvements in `code-block` and `literalinclude`. I think there is still a functionality that would be nice to add.

I think that it should be given an option to strip out lines containing a given string, pretty much the `:start-after` option does. It would allow for instance to remove special comments we do not want to enter the docs. 

My use case is the following: I extract code snippet form an example and some are nested in others, as

```
#snippet 1 begins
...
#snippet 2 begins
...
#snippet 2 ends
...
#snippet 1 ends
```

Ideally I would like to show snippet 2 to explain something specific, and then snippet 1 for the whole code. I think relying on code line is a fragile approach.

**Comment by andreacassioli** [comment]
I am surprise nobody commented. 

I think this is a quite general feature. For instance one may have
- comments that do not want to show
- remove empty  lines

A simple final strip would be very handy.

**Comment by atomgalaxy** [comment]
I need it for the same reason, would be super-useful.

**Comment by FernandoBasso** [comment]
Yeah, that would be nice to have. [Asciidoctor has a similar feature](https://docs.asciidoctor.org/asciidoc/latest/directives/include-lines/).

Sometimes it pains me that Markdown, and not Asciidoctor is mostly used for documentation and technical writing.

**Comment by tudortimi-ifx** [comment]
Would be useful for our team as well. We tried using comment markers like the @andreacassioli to implement what Asciidoctor calls [tagged regions](https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/). Had the same disadvantage that we can't strip them out when, for example, including the entire file.

**Comment by timhoffm** [comment]
More generally, would it make sense to support applying a user-defined filter function to the contents?
